### DISCUSSION TITLE
Disable Wi-Fi + BT

### DISCUSSION THREAD

**Discussion by askoa** [chipsec/chipsec#2176] [Q&A]
Is it possible to disable Wi-Fi and BT radios using Chipsec similar to what System76 and Dasharo does as documented [here](https://docs.dasharo.com/dasharo-menu-docs/dasharo-system-features/).

```
Enable Wi-Fi + BT radios - (applicable to laptops) - Enables or disables radios for Wi-Fi and Bluetooth. When disabled, the slot is unconfigured and power to the wireless module is disconnected.
```

I saw that tools like fiano removes files based on guid. Fiano works on firmware image on a file. I was looking for solution like Chipsec which can directly update the firmware. Also, I don't want to remove the FV file as I might want to enable it in the future. I am looking for a solution where I can disable Wi-Fi + BT radios.

Edit: the System76 firmware does this by setting some bit as per below link. Not sure if it can be translated to Chipsec command.

https://github.com/Dasharo/coreboot/blob/deb163591f95b794eac342dbdda7492554b69dbb/src/mainboard/clevo/tgl-u/variants/ns50mu/ramstage.c#L44

**Comment by npmitche** [comment]
So the way the System76 does it is it's sending the command 24 to the EC (embedded controller). You could potentially do this using chipsec: (to disable) `sudo python3 chipsec_util.py io write 0xe02 1 0 && sudo python3 chipsec_util.py io write 0xe00 1 24`

Chipsec util help: `chipsec_util io write <io_port> <width> <value>`
PortID found here: https://github.com/Dasharo/coreboot/blob/dasharo/src/ec/system76/ec/system76_ec.c#L62 (REG_CMD)
CMD Value found here: https://github.com/Dasharo/coreboot/blob/dasharo/src/ec/system76/ec/commands.h#L56

Please note, this is additional functionality that System76 has added to their EC. This will probably not work on other platforms unless those platforms have similar functionality.

**Reply by askoa** [reply]
Thanks for the response. I am trying to replicate the functionality in Lenovo and Macbook Pro  with Intel chip laptops. I believe both have proprietary firmware.

I worked out the guid for the fv files for the wifi driver.  I was wondering if I can use Chipsec to disable the radio.
